packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -230,40 +230,45 @@ class MacOSPlugin extends PluginPlatform {
 
 /// Contains the parameters to template a Windows plugin.
 ///
-/// The required fields include: [name] of the plugin, and [pluginClass] that will
-/// be the entry point to the plugin's native code.
+/// The [name] of the plugin is required. Either [dartPluginClass] or [pluginClass] are required.
+/// [pluginClass] will be the entry point to the plugin's native code.
 class WindowsPlugin extends PluginPlatform {
   const WindowsPlugin({
     @required this.name,
-    @required this.pluginClass,
+    this.pluginClass,
+    this.dartPluginClass,
   });
 
   factory WindowsPlugin.fromYaml(String name, YamlMap yaml) {
     assert(validate(yaml));
     return WindowsPlugin(
       name: name,
       pluginClass: yaml['pluginClass'] as String,
+      dartPluginClass: yaml['dartPluginClass'] as String,
     );
   }
 
   static bool validate(YamlMap yaml) {
     if (yaml == null) {
       return false;
     }
-    return yaml['pluginClass'] is String;
+    return (yaml['dartPluginClass'] != null && yaml['dartPluginClass'] is String) ||
```

Comment:
You shouldn't need the first parts of these checks; `null` is of type `Null`, not `String`. (Otherwise the code you are replacing wouldn't have been validating things correctly.)